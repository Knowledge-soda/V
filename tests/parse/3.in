:a, b
--(var) := =(var, +(v(var), -1))
+=(var, n) := =(var, +(v(var), n))
sum(x) := v(=(b, x) =(a, 0) rep: +=(a, v(b)) --(b) ijmp(v(b), rep) a)
print(sum(5))
