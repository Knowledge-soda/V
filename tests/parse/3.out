-1::
  BLN(NOP):
    BLN(PRNT):
      VAR(4):
        NUM(5), BLN(NOP),
      BLN(NOP),
    BLN(NOP),

2::
  BLN(MOV):
    VAL(0), COM, BLN(SUM):
      BLN(VAL):
        VAL(0), BLN(NOP),
      COM, NUM(-1), BLN(NOP),
    BLN(NOP),

3::
  BLN(MOV):
    VAL(0), COM, BLN(SUM):
      BLN(VAL):
        VAL(0), BLN(NOP),
      COM, VAL(1), BLN(NOP),
    BLN(NOP),

4::
  BLN(VAL):
    BLN(MOV):
      VAR(1), COM, VAL(0), BLN(NOP),
    BLN(MOV):
      VAR(0), COM, NUM(0), BLN(NOP),
    PLC(5), VAR(3):
      VAR(0), COM, BLN(VAL):
        VAR(1), BLN(NOP),
      BLN(NOP),
    VAR(2):
      VAR(1), BLN(NOP),
    BLN(IJMP):
      BLN(VAL):
        VAR(1), BLN(NOP),
      COM, VAR(5), BLN(NOP),
    VAR(0), BLN(NOP),

